### PR TITLE
feat: add api to get local function variant given function invocation

### PR DESCRIPTION
**Pull request by scgkiran**

### CODE REVIEW

File: functions/local_functions_test.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+	}{
+		{"avg:fp32", "function variant not found"},
+		{"avg:fp64", ""},
+		{"min:i8", "function variant not found"},
+		{"min:i16", ""},
+		{"min:i32", ""},
+		{"sum:i64", "function variant not found"},
+		{"sum:fp64", ""},
+		{"min:dec", ""},
+		{"min:dec", ""},
```
Comment (EpsilonPrime) [inline functions/local_functions_test.go:202]: Could we add a test with an incompatible option (similar to the test done for scalar above)?

File: testcases/parser/parse_test.go [hunk 2]
```diff
@@ -613,3 +616,57 @@ func TestParseTestCaseFile(t *testing.T) {
...
+			}
+		})
+	}
+}
+
+func testGetFunctionInvocation(t *testing.T, tc *TestCase, reg *expr.ExtensionRegistry, registry functions.FunctionRegistry) {
+	switch tc.FuncType {
+	case ScalarFuncType:
+		invocation, err := tc.GetScalarFunctionInvocation(reg, registry)
+		require.NoError(t, err, "GetScalarFunctionInvocation failed with error in test case: %s", tc.CompoundFunctionName())
```
Comment (EpsilonPrime) [inline testcases/parser/parse_test.go:651]: Is there a test case or documentation somewhere that shows what CompoundFunctionName does?  All I can gather from here is that it doesn't return an error sometimes.

File: testcases/parser/parse_test.go [hunk 3]
```diff
@@ -613,3 +616,57 @@ func TestParseTestCaseFile(t *testing.T) {
...
+		require.Equal(t, tc.ID().URI, invocation.ID().URI)
+	case AggregateFuncType:
+		invocation, err := tc.GetAggregateFunctionInvocation(reg, registry)
```
Comment (EpsilonPrime) [inline testcases/parser/parse_test.go:654]: Is there a CompoundFunctionName() for aggregates?

File: testcases/parser/parse_test.go (hunk 2 above)
Comment (scgkiran) [reply testcases/parser/parse_test.go:654]: It is similar to FunctionVariant.[CompoundName](https://github.com/substrait-io/substrait-go/blob/8e12e1e1ae704fb6235e803b9b6586504dc48c58/extensions/variants.go#L334-L334). It is a string with name and argTypes.

Added a test for TestCase.CompoundName. Earlier I was testing with ID()

File: testcases/parser/parse_test.go (hunk 3 above)
Comment (scgkiran) [reply testcases/parser/parse_test.go:657]: yes, fixed the error message in the assert.

**Review by EpsilonPrime — APPROVED** [review]

**Pull request merged**